tests/integration/cartography/intel/aws/test_dynamodb.py

```diff
@@ -1,33 +1,97 @@
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
 import cartography.intel.aws.dynamodb
 import tests.data.aws.dynamodb
+from tests.integration.util import check_nodes
+from tests.integration.util import check_rels
 
 TEST_ACCOUNT_ID = '000000000000'
 TEST_REGION = 'us-east-1'
 TEST_UPDATE_TAG = 123456789
 
 
-def test_load_dynamodb(neo4j_session):
-    data = tests.data.aws.dynamodb.LIST_DYNAMODB_TABLES['Tables']
+@patch.object(
+    cartography.intel.aws.dynamodb, 'get_dynamodb_tables',
+    return_value=tests.data.aws.dynamodb.LIST_DYNAMODB_TABLES['Tables'],
+)
+def test_load_dynamodb(mock_get_instances, neo4j_session):
+    """
+    Ensure that instances actually get loaded and have their key fields
+    """
+    # Arrange
+    boto3_session = MagicMock()
 
-    cartography.intel.aws.dynamodb.load_dynamodb_tables(
+    # Act
+    cartography.intel.aws.dynamodb.sync_dynamodb_tables(
         neo4j_session,
-        data,
+        boto3_session,
         TEST_REGION,
         TEST_ACCOUNT_ID,
         TEST_UPDATE_TAG,
+        {'UPDATE_TAG': TEST_UPDATE_TAG, 'AWS_ID': TEST_ACCOUNT_ID},
     )
-    expected_rows = 1000000
-    expected_nodes = {
-        ("arn:aws:dynamodb:us-east-1:000000000000:table/example-table", expected_rows),
-        ("arn:aws:dynamodb:us-east-1:000000000000:table/sample-table", expected_rows),
-        ("arn:aws:dynamodb:us-east-1:000000000000:table/model-table", expected_rows),
-        ("arn:aws:dynamodb:us-east-1:000000000000:table/basic-table", expected_rows),
+
+    # Assert ddb table nodes exist
+    assert check_nodes(neo4j_session, 'DynamoDBTable', ['id', 'rows']) == {
+        ("arn:aws:dynamodb:us-east-1:000000000000:table/example-table", 1000000),
+        ("arn:aws:dynamodb:us-east-1:000000000000:table/sample-table", 1000000),
+        ("arn:aws:dynamodb:us-east-1:000000000000:table/model-table", 1000000),
+        ("arn:aws:dynamodb:us-east-1:000000000000:table/basic-table", 1000000),
     }
```

Comment:
Can you add one more test to validate relationships between DynamoDBTable and the AWSAccount?

Reply:
And a few more tests to make sure that combinations of stale indexes and stale tables are properly cleaned up, since the way we're using the sub_resource_rel is different than how we've done it before.